packages/vue/src/pages/planning-scheduler.vue

```diff
@@ -0,0 +1,378 @@
+type SelectedAppointment = {
+  data: AppointmentData,
+  target: HTMLElement | undefined };
+
+const schedulerRef = ref<InstanceType<typeof DxScheduler>>();
+const tooltipRef = ref<InstanceType<typeof DxTooltip>>();
+const currentDate = ref(new Date());
+const currentView = ref('workWeek');
+const isRightPanelOpen = ref(false);
+
+const tasks = ref<DataSource | null>(null);
+
+const agendaItems = ref<AgendaItem[]>([]);
+
+const schedulerCurrentDate = computed((): Date => {
+  const schedulerInstance = schedulerRef?.value?.instance;
+  const startViewDate = schedulerInstance?.getStartViewDate();
+  const endViewDate = schedulerInstance?.getEndViewDate();
+
+  return (schedulerCurrentDate.value?.getMonth() !== currentDate.value?.getMonth()
+    || (startViewDate && startViewDate > currentDate.value)
+    || (endViewDate && endViewDate < currentDate.value)
+  )
+    ? currentDate.value
+    : schedulerCurrentDate.value;
+});
+
+const resourcesList: Record<string, unknown>[] = defaultCalendarListItems.reduce(
+  (res: Record<string, unknown>[], calendarList: {items: []}) => res.concat(calendarList.items),
+  [],
+);
+
+const selectedAppointment = ref<SelectedAppointment | null>(null);
+const rightPanelOpen = ref(false);
+
+const tooltipPosition = computed(() => {
+  if (screenInfo.value.isSmall) {
+    return 'bottom';
+  }
+
+  const appointmentTarget = selectedAppointment.value?.target;
+  const classList = (appointmentTarget as unknown as HTMLElement[])?.[0]?.classList
+    || appointmentTarget?.classList;
+
+  return classList?.contains('dx-list') && rightPanelOpen ? 'left' : 'top';
+});
+
+watchEffect(() => {
+  const deps = [tasks.value, currentView.value, currentDate.value, isRightPanelOpen.value];
+  if (deps.length) {
+    setTimeout(() => schedulerRef.value?.instance?.repaint());
+  }
+});
+
+getTasksForScheduler().then((data: Task[]) => {
+  tasks.value = new DataSource(data);
+});
+
+function toggleRightPanelOpen(isOpen?: boolean) {
+  isRightPanelOpen.value = isOpen || !isRightPanelOpen.value;
+}
+
+function findAllAppointmentsForDay(selectedAppointmentData?: AppointmentData) {
+  if (!tasks.value) {
+    return [];
+  }
+  const appointments = tasks.value.items();
+  if (appointments.length === 0 || !selectedAppointmentData) {
+    return [];
+  }
+  return appointments.filter(
+    (appointment) => appointment.startDate.getDate() === selectedAppointmentData.startDate.getDate()
+        && appointment.startDate.getMonth() === selectedAppointmentData.startDate.getMonth(),
+  );
+}
+
+function updateAgenda(appointmentData?: AppointmentData) {
+  agendaItems.value = findAllAppointmentsForDay(appointmentData);
+}
+
+function onCalendarDateChange({ value }: { value: Date }) {
+  currentDate.value = value;
+  updateAgenda({ startDate: currentDate.value });
+}
+
+function showAppointmentCreationForm(appointment: SelectedAppointment| null) {
+  schedulerRef.value?.instance.showAppointmentPopup(appointment?.data, !appointment);
```

Comment:
Should this form actually create an appointment? It doesn't create one for me...

Reply:
fixed